core/services/validators.go

```diff
@@ -382,8 +385,59 @@ func ValidateServiceAgreement(sa models.ServiceAgreement, store *store.Store) er
 	return fe.CoerceEmptyToNil()
 }
 
-// ValidatedOracleSpec validates an oracle spec that came from TOML
-func ValidatedOracleSpec(tomlString string) (offchainreporting.OracleSpec, error) {
+var (
+	ErrNoSuchPeerID             = errors.New("no such peer id exists")
+	ErrNoSuchKeyBundle          = errors.New("no such key bundle exists")
+	ErrNoSuchTransmitterAddress = errors.New("no such transmitter address exists")
+)
+
+func ValidateOracleSpec(tomlString string, db *gorm.DB) (offchainreporting.OracleSpec, error) {
```

Comment:
I'm kind of surprised this is necessary, isn't it automatically caught by the foreign keys at insertion time?

Reply:
yeah FK constraints catch it, but the point is to improve the UX and not throw a raw pg error at the user. More details in the ticket description

Reply:
There's a downside to this strategy in that it can non deterministically fail if enough people are using the front end at the same time, usually it's better to try and match on the consistency error, I dunno though I'm allergict to non determinism.

Reply:
pushed it down to the job.ORM layer so its all in the same tx